### PR TITLE
Cleanup, add docBlocks, improve tests

### PR DESCRIPTION
**Pull request by devanych**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,39 +15,55 @@ The package allows responding with data that is automatically converted into PSR
 [![static analysis](https://github.com/yiisoft/data-response/workflows/static%20analysis/badge.svg)](https://github.com/yiisoft/data-response/actions?query=workflow%3A%22static+analysis%22)
 [![type-coverage](https://shepherd.dev/github/yiisoft/data-response/coverage.svg)](https://shepherd.dev/github/yiisoft/data-response)
 
+The package allows responding with data that is automatically converted into PSR-7 response.
+
+## Installation
```
Comment (vjik) [inline README.md:20]: Also need section "Requirements" with PHP version and extensions. For example: https://github.com/yiisoft/translator-formatter-intl#requirements

File: src/DataResponse.php [hunk 2]
```diff
@@ -116,10 +140,12 @@ public function hasHeader($name): bool
     }
 
     /**
+     * @inheritDoc
+     *
      * @param string $name
      * @param string|string[] $value
      *
-     * @return static
+     * @return self
```
Comment (vjik) [inline src/DataResponse.php:148]: Can be removed, becuase don't add more info.

File: src/DataResponse.php [hunk 3]
```diff
@@ -130,7 +156,9 @@ public function withAddedHeader($name, $value): self
     }
 
     /**
-     * @return static
+     * @inheritDoc
+     *
+     * @return self
```
Comment (vjik) [inline src/DataResponse.php:161]: Can be removed, becuase don't add more info.

File: src/DataResponse.php [hunk 4]
```diff
@@ -142,10 +170,12 @@ public function withBody(StreamInterface $body): self
     }
 
     /**
+     * @inheritDoc
+     *
      * @param string $name
      * @param string|string[] $value
      *
-     * @return static
+     * @return self
```
Comment (vjik) [inline src/DataResponse.php:178]: Can be removed, becuase don't add more info.

File: src/DataResponse.php [hunk 5]
```diff
@@ -156,9 +186,11 @@ public function withHeader($name, $value): self
     }
 
     /**
+     * @inheritDoc
+     *
      * @param string $name
      *
-     * @return static
+     * @return self
```
Comment (vjik) [inline src/DataResponse.php:193]: Can be removed, becuase don't add more info.

File: src/DataResponse.php [hunk 6]
```diff
@@ -169,9 +201,11 @@ public function withoutHeader($name): self
     }
 
     /**
+     * @inheritDoc
+     *
      * @param string $version
      *
-     * @return static
+     * @return self
```
Comment (vjik) [inline src/DataResponse.php:208]: Can be removed, becuase don't add more info.

File: src/DataResponse.php [hunk 7]
```diff
@@ -182,10 +216,12 @@ public function withProtocolVersion($version): self
     }
 
     /**
+     * @inheritDoc
+     *
      * @param int $code
      * @param string $reasonPhrase
      *
-     * @return static
+     * @return self
```
Comment (vjik) [inline src/DataResponse.php:224]: Can be removed, becuase don't add more info.

File: src/DataResponse.php [hunk 8]
```diff
@@ -196,54 +232,88 @@ public function withStatus($code, $reasonPhrase = ''): self
     }
 
     /**
-     * @return static
+     * Returns a new instance with the specified response formatter.
+     *
+     * @param DataResponseFormatterInterface $responseFormatter
+     *
+     * @return self
```
Comment (vjik) [inline src/DataResponse.php:239]: Can be removed, becuase don't add more info.

File: src/DataResponse.php [hunk 9]
```diff
@@ -196,54 +232,88 @@ public function withStatus($code, $reasonPhrase = ''): self
...
         return $this->responseFormatter !== null;
     }
 
-    public function getResponse(): ResponseInterface
+    /**
+     * Returns a new instance with the specified response data.
+     *
+     * @param mixed $data The response data.
+     *
+     * @return self
```
Comment (vjik) [inline src/DataResponse.php:274]: Can be removed, becuase don't add more info.

File: src/Middleware/ContentNegotiator.php [hunk 10]
```diff
@@ -13,23 +13,40 @@
...
     }
 
     /**
-     * @param array $contentFormatters
+     * Returns a new instance with the specified content formatters.
+     *
+     * @param array<string, DataResponseFormatterInterface> $contentFormatters The array key is the content type,
+     * and the value is the {@see DataResponseFormatterInterface} instance.
+     *
+     * @return self
```
Comment (vjik) [inline src/Middleware/ContentNegotiator.php:49]: Can be removed, becuase don't add more info.

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/DataResponse.php (hunk 2 above)
Comment (devanych) [reply src/DataResponse.php:148]: Cannot be removed. The Psalm fails.

File: src/DataResponse.php (hunk 3 above)
Comment (devanych) [reply src/DataResponse.php:161]: Cannot be removed. The Psalm fails.

File: src/DataResponse.php (hunk 4 above)
Comment (devanych) [reply src/DataResponse.php:178]: Cannot be removed. The Psalm fails.

File: src/DataResponse.php (hunk 5 above)
Comment (devanych) [reply src/DataResponse.php:193]: Cannot be removed. The Psalm fails.

File: src/DataResponse.php (hunk 6 above)
Comment (devanych) [reply src/DataResponse.php:208]: Cannot be removed. The Psalm fails.

File: src/DataResponse.php (hunk 7 above)
Comment (devanych) [reply src/DataResponse.php:224]: Cannot be removed. The Psalm fails.

File: src/DataResponse.php (hunk 8 above)
Comment (devanych) [reply src/DataResponse.php:239]: Cannot be removed. The Psalm fails.

File: src/DataResponse.php (hunk 9 above)
Comment (devanych) [reply src/DataResponse.php:274]: Cannot be removed. The Psalm fails.

File: src/Middleware/ContentNegotiator.php (hunk 10 above)
Comment (devanych) [reply src/Middleware/ContentNegotiator.php:49]: Cannot be removed. The Psalm fails.

File: README.md (hunk 1 above)
Comment (devanych) [reply README.md:25]: Done.

File: src/DataResponseFactoryInterface.php [hunk 11]
```diff
@@ -6,7 +6,19 @@
 
 use Yiisoft\Http\Status;
 
+/**
+ * DataResponseFactoryInterface is the interface that should be implemented data response factory classes.
```
Comment (samdark) [inline src/DataResponseFactoryInterface.php:10]:
```suggestion
 * DataResponseFactoryInterface is the interface that should be implemented by data response factory classes.
```

File: src/DataResponseFormatterInterface.php [hunk 12]
```diff
@@ -6,7 +6,17 @@
 
 use Psr\Http\Message\ResponseInterface;
 
+/**
+ * DataResponseFormatterInterface is the interface that should be implemented data response formatter classes.
```
Comment (samdark) [inline src/DataResponseFormatterInterface.php:10]:
```suggestion
 * DataResponseFormatterInterface is the interface that should be implemented by data response formatter classes.
```

File: src/Formatter/HtmlDataResponseFormatter.php [hunk 13]
```diff
@@ -17,6 +17,9 @@
 use function method_exists;
 use function sprintf;
 
+/**
+ * HtmlDataResponseFormatter formats the response data in HTML.
```
Comment (samdark) [inline src/Formatter/HtmlDataResponseFormatter.php:21]:
```suggestion
 * HtmlDataResponseFormatter formats the response data as HTML.
```

File: src/Formatter/JsonDataResponseFormatter.php [hunk 14]
```diff
@@ -10,6 +10,9 @@
 use Yiisoft\DataResponse\ResponseContentTrait;
 use Yiisoft\Json\Json;
 
+/**
+ * JsonDataResponseFormatter formats the response data in JSON.
```
Comment (samdark) [inline src/Formatter/JsonDataResponseFormatter.php:14]:
```suggestion
 * JsonDataResponseFormatter formats the response data as JSON.
```

File: src/Formatter/XmlDataResponseFormatter.php [hunk 15]
```diff
@@ -20,6 +20,9 @@
 use function is_int;
 use function is_object;
 
+/**
+ * XmlDataResponseFormatter formats the response data in XML.
```
Comment (samdark) [inline src/Formatter/XmlDataResponseFormatter.php:24]:
```suggestion
 * XmlDataResponseFormatter formats the response data as XML.
```

**Comment by samdark** [comment]
👍

**Pull request merged**